### DISCUSSION TITLE
Make core Syntax groups more semantically concise?

### DISCUSSION THREAD

**Discussion by adrian5** [2giosangmitom/nightfall.nvim#67] [General]
I have some thoughts about the coloring of the vim syntax groups. I don't know which languages you mainly use, but I think they could see some adjustment.

The groups as shown via `:h group-name`:

![group-name](https://github.com/user-attachments/assets/2610aff9-3edd-4c71-a8f7-6f00d0d31f91)

* `Character` is for standalone chars like `'a'`, not escape characters in strings. It should link to `String` imo.
* Where does `Statement` find use? I think semantically it should be pink instead of purple. The groups beneath it will inherit that by default.
* I would make `Repeat` and maybe even `Label` default-inherit from `Statement`. It makes sense in terms of their function.
* If `Macro` is to be like `Function`, I would link it to the latter to emphasize that choice.
* Consider if you want a different color for the `PreProc` cluster, now that some colors are liberated. Though pink looks fine there too.
* `Type` being the same as `String` could be a bit confusing, but here I understand the need for being colorful. An alternative would be one of the blue hues.
* `Typedef` should be the same as `Type` imo, it's often used for new types, e.g. `dVec` in C++ `using dVec = std::vector<double>;`
* Is `Special` good as yellow? I don't know, I encounter it so rarely to form an opinion.
* `SpecialChar` is escape characters like `\n` and `%d` in `print("The number %d is great!\n")`. For Treesitter you used `["@string.escape"] = { fg = colors.light_cyan }` and I think that gives better contrast than the current `lavender`.
* `Todo` should probably match the Treesitter `["@comment.todo"] = { fg = colors.black, bg = colors.turquoise }`. Which color do you prefer? Turquoise seems fine.

Let me know your thoughts on these when you find the time. I could prepare changes in a PR later on.

**Comment by 2giosangmitom** [comment]
I fully agree with your suggestions. I primarily utilize JavaScript and C++ in my development work, and I believe this is a fantastic adjustment.
